lib/cpp/antaris_api_pyfunctions.cc

```diff
@@ -0,0 +1,197 @@
+        exit_status = An_GENERIC_FAILURE;
+        goto cleanup_and_exit;
+    }
+
+    g_File_String = cJSON_GetStringValue(pJsonStr);
+    if (g_File_String == NULL) {
+        printf("Error: %s string not present \n", JSON_Key_File_Conn);
+    }
+
+    // get Truetwin_Dir
+    pJsonStr = cJSON_GetObjectItem(key_ftm, JSON_Key_TrueTwin_Dir);
+    if (pJsonStr == NULL) {
+        printf("Error: %s key absent in config.json \n", JSON_Key_TrueTwin_Dir);
+        exit_status = An_GENERIC_FAILURE;
+        goto cleanup_and_exit;
+    }
+
+    if (cJSON_IsString(pJsonStr) == cJSON_Invalid) {
+        printf("Error: %s value is not a string \n", JSON_Key_TrueTwin_Dir);
+        exit_status = An_GENERIC_FAILURE;
+        goto cleanup_and_exit;
+    }
+
+    g_Truetwin_Dir = cJSON_GetStringValue(pJsonStr);
+    if (g_Truetwin_Dir == NULL) {
+        printf("Error: %s string not present \n", JSON_Key_TrueTwin_Dir);
+    }
+
+    // get  Share_Name
+    pJsonStr = cJSON_GetObjectItem(key_ftm, JSON_Key_Share_Name);
+    if (pJsonStr == NULL) {
+        printf("Error: %s key absent in config.json \n", JSON_Key_Share_Name);
+        exit_status = An_GENERIC_FAILURE;
+        goto cleanup_and_exit;
+    }
+
+    if (cJSON_IsString(pJsonStr) == cJSON_Invalid) {
+        printf("Error: %s value is not a string \n", JSON_Key_Share_Name);
+        exit_status = An_GENERIC_FAILURE;
+        goto cleanup_and_exit;
+    }
+
+    g_Share_Name = cJSON_GetStringValue(pJsonStr);
+    if (g_Share_Name == NULL) {
+        printf("Error: %s string not present \n", JSON_Key_Share_Name);
+        exit_status = An_GENERIC_FAILURE;
+        goto cleanup_and_exit;
+    }
+
+    // Creating destination file path
+    position = strstr(download_file_params->file_path, FILE_DOWNLOAD_DIR);
+    if ( position == NULL ) {
+        printf("Error: source file should be at %s \n", FILE_DOWNLOAD_DIR);
+        exit_status = An_GENERIC_FAILURE;
+        goto cleanup_and_exit;
+    }
+
+    position += strlen(FILE_DOWNLOAD_DIR);
+    remaining_length = strlen(position);
+    strcpy(dst_file_name, g_Truetwin_Dir);
+    strncat(dst_file_name, position, remaining_length);
+
+    pArgs = PyTuple_New(4);  // Create a tuple with 4 elements
+
+    PyTuple_SetItem(pArgs, 0, Py_BuildValue("s", download_file_params->file_path));  // 's' for string
+    PyTuple_SetItem(pArgs, 1, Py_BuildValue("s", g_File_String));
+    PyTuple_SetItem(pArgs, 2, Py_BuildValue("s", g_Share_Name));
+    PyTuple_SetItem(pArgs, 3, Py_BuildValue("s", dst_file_name));
+
+    printf("Antaris_SatOS : Uploading file %s \n", download_file_params->file_path);
+
+    pName = PyUnicode_DecodeFSDefault(PYTHON_SCRIPT_FILE);
+    pModule = PyImport_Import(pName);
+    if (pModule == nullptr)   {
+        printf("Error: Import failed, Can not upload file %s \n", download_file_params->file_path);
+        exit_status = An_GENERIC_FAILURE;
+        goto cleanup_and_exit;
```

Comment:
For all error cases, we are redirecting to same label. But in some cases like from here, pFunction and pResults will not be initialized. Will Py_XDECREF() will take care of such case where the pointer passed to it is NULL?